internal/framework/generic_resource.go

```diff
@@ -310,13 +313,34 @@ func (r *GenericResource) GetTimeout(operation string, defaultTimeout time.Durat
 	return defaultTimeout
 }
 
+type resourceFailed struct {
```

Comment:
Nit: Can you name the error `resourceFailedError`?

Reply:
Will do.